history_test.go

```diff
@@ -0,0 +1,127 @@
+package merkletree
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var signKey, _ = crypto.GenerateKey()
+
+// scenario:
+// 1st: epoch = 1
+// 2nd: epoch = 3
+// 3nd: epoch = 5 (latest STR)
+func TestHistoryHashChain(t *testing.T) {
+	var startupTime int64
+	var epochInterval int64
+
+	startupTime = 1
+	epochInterval = 2
+
+	key1 := "key"
+	val1 := []byte("value")
+
+	key2 := "key2"
+	val2 := []byte("value2")
+
+	key3 := "key3"
+	val3 := []byte("value3")
+
+	m1 := InitMerkleTree(&DefaultPolicies{}, treeNonce)
```

Comment:
It's kind of confusing that `treeNonce` is declared and defined in another `_test.go` file.